src/DynamoCore/DSEngine/NodeToCode.cs

```diff
@@ -738,6 +738,9 @@ public string GetNextName()
                         var inputVar = node.GetAstIdentifierForOutputIndex(i).Value;
                         outputMap[inputVar] = string.Empty;
                     }
+
+                    var previewVar = node.AstIdentifierForPreview.Value;
+                    outputMap[previewVar] = string.Empty;
```

Comment:
@Benglin , yes. For single output node, `AstIdentifiderForPreview` is the output variable (from output prot), but for multi output node it isn't, so it become an intermediate variable.

Ideally, I was thinking to add a `NodeModel.GetIntermediateVariables()` virtual function so that any node is free to create many intermediate variables, but it is a little bit heavy now and I have concern to get n2c stuff into basic class. If there are more cases about intermediate variables, then I'll do that.